plonky3/src/circuit_builder.rs

```diff
@@ -193,11 +135,12 @@ impl<'a, T: FieldElement> PowdrCircuit<'a, T> {
     }
 
     /// Conversion to plonky3 expression
-    fn to_plonky3_expr<AB: AirBuilder<F = Val>>(
+    fn to_plonky3_expr<AB: AirBuilder<F = Val> + AirBuilderWithPublicValues>(
         &self,
         e: &AlgebraicExpression<T>,
         main: &AB::M,
         fixed: &AB::M,
+        publics: &Vec<<AB as AirBuilderWithPublicValues>::PublicVar>,
```

Comment:
It would be clearer for this to be `&HashMap<String, <AB as AirBuilderWithPublicValues>::PublicVar>`. Basically a map of the values of the public inputs, indexed by name.